File: frontend/src/types/item.ts
export interface Item {
  id: number
  title: string
  description: string | null
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface ItemCreate {
  title: string
  description?: string
  is_active?: boolean
}

export interface ItemUpdate {
  title?: string
  description?: string
  is_active?: boolean
}

export interface ItemList {
  items: Item[]
  total: number
  page: number
  page_size: number
}

File: backend/app/schemas/item.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ItemBase(BaseModel):
    title: str
    description: str | None = None
    is_active: bool = True


class ItemCreate(ItemBase):
    pass


class ItemUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    is_active: bool | None = None


class ItemResponse(ItemBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime
    updated_at: datetime


class ItemList(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    items: list[ItemResponse]
    total: int
    page: int
    page_size: int